### PR TITLE
Ardak hw5

### PR DESCRIPTION
**Pull request by ArdakHere**

### CODE REVIEW

**Review by galetskaya-quantori — COMMENTED** [review]
Great job! You have well-structured and easy to read app. It's a good practice that you've used docstrings to describe your endpoints.
Here are some suggestions for improvements:
Using `id` instead of `identifier` is more conventional and concise.
Ensure consistent spacing between endpoints and functions. It's a good practice to use 2 blank lines between top-level definitions for enhanced readability.
Remember to add cache files to your .gitignore to keep your repository clean.
I have also added some comments on the PR

File: src/main.py [hunk 1]
```diff
@@ -1,18 +1,196 @@
...
+        \n str: "Molecule added successfully" or
+        \n "The molecule you are trying to add already exists in the database."
+        or "Can't duplicate identifiers!" or "Invalid SMILES, try again"
+    """
+    if not is_valid_smiles(molecule.dict()['smiles']):
+        raise HTTPException(status_code=400, detail="Invalid SMILES, try again")
+
+    if molecules_db.__len__() != 0:
+        for item in molecules_db:
+            if molecule.dict()['identifier'] in unique_identifiers:
```
Comment (galetskaya-quantori) [inline src/main.py:30]: You can simplify it and use 
`if molecule.identifier in unique_identifiers:`

File: src/main.py [hunk 2]
```diff
@@ -1,18 +1,196 @@
...
+    """
+    if identifier != updated_molecule.dict()['identifier']:
+        return "Can't change the identifier of the molecule"
+    if not is_valid_smiles(updated_molecule.dict()['smiles']):
+        return "Invalid SMILES string, try again"
+    if updated_molecule in molecules_db:
+        return "The molecule you are trying to change already exists in the database."
+
+
+    for index, user in enumerate(molecules_db):
```
Comment (galetskaya-quantori) [inline src/main.py:81]: Using `user` as the variable name here is a bit confusing,

Comment (ArdakHere) [reply src/main.py:81]: Oh, my bad. I was using your way of iterating through the dict you used on your second lecture. Should have used "molecule" though

**Pull request merged**